### PR TITLE
Add pagination of long collections to inspector

### PR DESCRIPTION
**Pull request by alexander-yakushev**
Let's get this train going. Previous discussion is at #123. This pull request handles the behind-the-scenes part of the feature.

Sister pull request is clojure-emacs/cider#1212.

### CODE REVIEW

File: src/cider/nrepl/middleware/inspect.clj [hunk 1]
```diff
@@ -87,4 +112,7 @@
    :expects #{"eval"}
    :handles {"inspect-pop" {}
              "inspect-push" {}
-             "inspect-refresh" {}}}))
+             "inspect-refresh" {}
```
Comment (bbatsov) [inline src/cider/nrepl/middleware/inspect.clj]: Btw, while you're on this you can add the missing metadata here (op description & params).

File: test/clj/cider/nrepl/middleware/inspect_test.clj [hunk 2]
```diff
@@ -91,6 +98,77 @@
...
+    (is (= 21 (-> long-sequence
+                  inspect
+                  (inspect/set-page-size 20)
+                  :counter)))
+    (is (= '(:newline) (-> long-sequence
+                           inspect
+                           (inspect/set-page-size 200)
+                           :rendered
+                           last))))
+  (testing "uncounted collections have their size figured on the last page"
```
Comment (bbatsov) [inline test/clj/cider/nrepl/middleware/inspect_test.clj]: figured -> figured out (or just determined)

**Comment by bbatsov** [comment]
The changes look good. I've added some tiny remarks and obviously we need to get those failing tests to pass.

**Comment by alexander-yakushev** [comment]
> Btw, while you're on this you can add the missing metadata here (op description & params).
Exploiter!

**Comment by alexander-yakushev** [comment]
RE: tests. Turns out, in 1.7 `(range 100)` is not a `LazySeq`, but a `LongRange`. Huh.

**Comment by alexander-yakushev** [comment]
@bbatsov. Regarding eastwood: jonase/eastwood#169. Regarding formatting: cljfmt and clojure-mode don't agree on how to align `as->`.

**Comment by bbatsov** [comment]
Guess you can just turn off this eastwood check for now. I believe `as->` is correctly formatted by `clojure-mode`. If you agree submit an issue report to `cljfmt`, if not - to `clojure-mode`. :-)

**Comment by alexander-yakushev** [comment]
I don't have a preference, I just want to merge this stupid thing :D.

**Comment by Malabarba** [comment]
Report to cljfmt. I had the same issue with cond-> and they fixed it within the hour.

**Comment by alexander-yakushev** [comment]
OK, I greened all the things, where's my cookie? :stuck_out_tongue_winking_eye:

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**